### DISCUSSION TITLE
interpolate resonant signal varying scale and resolution

### DISCUSSION THREAD

**Discussion by wiso** [scikit-hep/pyhf#1732] [Q&A]
Hello, I am wondering how easy/feasible is to parameterize a resonant signal as a function of the scale (e.g. the particle mass) and the resolution. I cannot simply redo the analysis with a different signal model, since the scale and resolution systematic is important.

For example I am wondering if something like this is implemented:

https://arxiv.org/pdf/1410.7388.pdf
https://root.cern.ch/doc/master/classRooMomentMorph.html

**Comment by kratsg** [comment]
So I'll add @alexander-held and @cranmer but we've discussed morphing as being included in a `v2` HistFactory spec, but this is not something that exists in `v1`. There are cases where a horizontal interpolation could be well motivated (I think this is what you're asking for).

Right now, we're still on `v1`, but ideas and future improvements/features like these are being collected for a `v2` spec (no central location yet afaik) but that's why I tagged those two folx.

**Reply by wiso** [reply]
Thanks @kratsg, horizontal interpolation is what I need. Not sure what is v1/v2, I've understood this is not implemented in pyhf right now.

**Reply by kratsg** [reply]
So v1 refers to "1.0.0" of the [HistFactory JSON schema](https://scikit-hep.org/pyhf/schemas/1.0.0/workspace.json) which is roughly equivalent to the corresponding [HistFactory XML schema](https://scikit-hep.org/pyhf/schemas/HistFactorySchema.dtd) (although the latter is not versioned as explicitly). A version 2 of this schema would be a major breaking change that is not necessarily backwards-compatible [e.g. you might not be able to go from v2 to v1, but you should be able to go from v1 to v2], and would tentatively include additional things such as morphing.
